commit 84fa69fc4689ff02afa1a0d868af847feb32f556 (refs/changes/31/131/1)
Author: Dave Pacheco <dap@joyent.com>
Date:   2016-07-22T10:15:27-07:00 (3 years, 3 months ago)
    
    TOOLS-1496 add link to cr.joyent.us docs on cr.joyent.us

diff --git a/images/appserver/Dockerfile b/images/appserver/Dockerfile
index 9e6bc32..22e8c6a 100644
--- a/images/appserver/Dockerfile
+++ b/images/appserver/Dockerfile
@@ -90,6 +90,7 @@ RUN  set -o xtrace && \
 #
 COPY ./etc/gerrit.config.base ${GERRIT_HOME}/shipped/gerrit.config.base
 COPY ./etc/replication.config ${GERRIT_HOME}/shipped/replication.config
+COPY ./theme ${GERRIT_HOME}/shipped/theme
 
 #
 # Configure the entrypoint (which runs every time) and the default start
diff --git a/images/appserver/bin/gerrit-entrypoint.sh b/images/appserver/bin/gerrit-entrypoint.sh
index 497e675..c281119 100644
--- a/images/appserver/bin/gerrit-entrypoint.sh
+++ b/images/appserver/bin/gerrit-entrypoint.sh
@@ -200,6 +200,21 @@ function main
 		ln -s $GERRIT_SITE/ssh-joyent $GERRIT_HOME/.ssh
 	fi
 
+	#
+	# These files should be symlinks in the volume container, but were never
+	# in the container before, so don't need to be archived like the ones
+	# above.
+	#
+	if ! [[ -L "$GERRIT_SITE"/etc/GerritSite.css ]]; then
+		ln -s "$GERRIT_HOME/shipped/theme/GerritSite.css" \
+		    "$GERRIT_SITE/etc/GerritSite.css"
+	fi
+
+	if ! [[ -L "$GERRIT_SITE"/etc/GerritSiteHeader.html ]]; then
+		ln -s "$GERRIT_HOME/shipped/theme/GerritSiteHeader.html" \
+		    "$GERRIT_SITE/etc/GerritSiteHeader.html"
+	fi
+
 	#
 	# Historically, the Gerrit user's numeric uid changed across different
 	# image versions, though we've now settled on one that's encoded in the
diff --git a/images/appserver/theme/GerritSite.css b/images/appserver/theme/GerritSite.css
new file mode 100644
index 0000000..63dfb71
--- /dev/null
+++ b/images/appserver/theme/GerritSite.css
@@ -0,0 +1,12 @@
+/*
+ * See the Gerrit documentation for information about this file.
+ */
+table.crjHeader {
+	width: 100%;
+	position: fixed;
+	right: 5px;
+	top: 35px;
+
+	font-weight: bold;
+	text-align: right;
+}
diff --git a/images/appserver/theme/GerritSiteHeader.html b/images/appserver/theme/GerritSiteHeader.html
new file mode 100644
index 0000000..0409650
--- /dev/null
+++ b/images/appserver/theme/GerritSiteHeader.html
@@ -0,0 +1,9 @@
+<table class="crjHeader">
+<!--
+    See the Gerrit documentation for constraints on the contents of this file.
+-->
+<tr>
+<td><a
+href="https://github.com/joyent/joyent-gerrit/blob/master/docs/user/README.md">New User Guide</a></td>
+</tr>
+</table>
